### ISSUE TITLE
SSE client gets empty events out of nowhere

### ISSUE THREAD

**Issue by Muddz** [r3labs/sse#72]
I have made an Windows program in Golang which uses SSE (the client part) and somehow the SSE client gets empty events out of nowhere. They don't originate from the server because I have checked the logs of both the Windows client program and the Server program. 

I managed to log the SSE event data when I get these empty events and got the following:

```
2020/01/21 20:20:42 SSE EVENT DATA: []
2020/01/21 20:20:42 SSE EVENT ID: []
2020/01/21 20:20:42 SSE EVENT RETRY: []
```

**The following code is the from the Client**

```
func connectToSSE() {
	sseClient.ReconnectStrategy = getBackOff()
	err := sseClient.Subscribe("sseID", func(msg *sse.Event) {
		b, _ := base64.StdEncoding.DecodeString(string(msg.Data))
		go saveTodoFile(string(b))
	})

	if err != nil {
		log.Println("Error: couldn't subscribe to SSE:", err)
	}
}
```

```
func getBackOff() *backoff.ExponentialBackOff {
	var customBackOff = backoff.NewExponentialBackOff()
	customBackOff.MaxElapsedTime = 0            //try until forever time
	customBackOff.MaxInterval = 2 * time.Second //no more than 2 seconds till next retry
	customBackOff.Multiplier = 1                //Generate the same delay between retries
	return customBackOff
}
```

**Comment by Muddz** [comment]
I also have no idea how to reproduce these this. But I have one thing I suspect. It may happen when I Logout/Login in Windows, but I tried that and couldn't force a reproduction of this event

**Comment by purehyperbole** [comment]
It's tough to say what the issue is from what you've describe. Are you seeing many of these events? Are they followed by non-empty events?

Could you please try with the following option?

This should error the connection if nothing is received from the server in a given time.

```go
sseClient := sse.NewClient(url)

sseClient.Connection.Transport = .&http.Transport{
        Dial: (&net.Dialer{
                Timeout:   30 * time.Second,
                KeepAlive: 30 * time.Second,
        }).Dial
}
```

**Comment by Muddz** [comment]
Hi again. I got the Empty event again "randomly".

```
2020/01/24 15:26:04 SSE EVENT DATA:
 []
2020/01/24 15:26:04 SSE EVENT ID:
 []
2020/01/24 15:26:04 SSE EVENT RETRY:
[]
```

I was actually using my Windows Go client saround 15:30 and later, but I didn't see any empty data. 
But around 19:20 the Client suddenly showed empty data. Its like the Empty event was dispatched at 15:26 but was first applied on the client many ours later.

I checked the Windows Log to see what happend in Windows at `15:26:04`. 

**At 15:24:** Windows reports WHEA-Logger Code 17 PCI-Express Hardware correction

**15:26**: At the time of the empty SSE event windows has not logged anything at all!
Only at 15:26 is no Windows or System events logged, anything before and after is. 
 
**At 16:53:** Windows went to hibernation mode

**At 19:17:** Windows woke up from Hibernation mode

**At 19:20:** Go Windows Client suddenly shows empty data
